roles/lib_utils/action_plugins/sanity_checks.py

```diff
@@ -219,6 +219,35 @@ def check_unsupported_nfs_configs(self, hostvars, host):
                     ''.format(storage))
         return None
 
+    def check_htpasswd_provider(self, hostvars, host):
+        """Fails if openshift_master_identity_providers contains an entry of
+        kind HTPasswdPasswordIdentityProvider and
+        openshift_master_manage_htpasswd is False"""
+
+        idps = self.template_var(
+            hostvars, host, 'openshift_master_identity_providers')
+        if not idps:
+            # If we don't find any identity_providers, nothing for us to do.
+            return None
+        manage_pass = self.template_var(
+            hostvars, host, 'openshift_master_manage_htpasswd')
+        if to_bool(manage_pass):
```

Comment:
Should we fail here too if the keys are present and just tell them to remove the keys, we'll move the file if doing an upgrade?